pkg/splunk/controller/util.go

```diff
@@ -140,6 +140,17 @@ func MergePodSpecUpdates(current *corev1.PodSpec, revised *corev1.PodSpec, name
 			"revised", len(revised.InitContainers))
 		current.InitContainers = revised.InitContainers
 		result = true
+	} else {
+		for idx := range current.InitContainers {
+			// check Image
+			if current.InitContainers[idx].Image != revised.InitContainers[idx].Image {
+				scopedLog.Info("Init Container Images differ",
+					"current", current.InitContainers[idx].Image,
+					"revised", revised.InitContainers[idx].Image)
+				current.InitContainers[idx].Image = revised.InitContainers[idx].Image
+				result = true
+			}
```

Comment:
Should we also check for differences in other fields such as `ports`, `resources`, `env variables` etc on the init container?

Reply:
Right now, we are not allowing any of these parameters(ports/resources, env, etc) through the CR. Image is the only field that we are taking from the config parameter, so, addressing that selectively for now.